code/game/objects/structures/crates_lockers/closets/statue.dm

```diff
@@ -1,146 +1,106 @@
-/obj/structure/closet/statue
+/obj/structure/statue/petrified
 	name = "statue"
 	desc = "An incredibly lifelike marble carving."
-	icon = 'icons/obj/statue.dmi'
 	icon_state = "human_male"
 	density = 1
 	anchored = 1
 	obj_integrity = 200
 	max_integrity = 200
-	var/intialTox = 0 	//these are here to keep the mob from taking damage from things that logically wouldn't affect a rock
-	var/intialFire = 0	//it's a little sloppy I know but it was this or the GODMODE flag. Lesser of two evils.
-	var/intialBrute = 0
-	var/intialOxy = 0
 	var/timer = 240 //eventually the person will be freed
+	var/mob/living/petrified_mob
 
-/obj/structure/closet/statue/New(loc, var/mob/living/L)
-
-	if(ishuman(L) || ismonkey(L) || iscorgi(L))
+/obj/structure/statue/petrified/New(loc, mob/living/L, statue_timer)
+	if(statue_timer)
+		timer = statue_timer
+	if(L)
+		petrified_mob = L
 		if(L.buckled)
 			L.buckled.unbuckle_mob(L,force=1)
-		L.reset_perspective(src)
-		L.loc = src
-		L.disabilities += MUTE
-		L.faction += "mimic" //Stops mimics from instaqdeling people in statues
 		L.visible_message("<span class='warning'>[L]'s skin rapidly turns to marble!</span>", "<span class='userdanger'>Your body freezes up! Can't... move... can't...  think...</span>")
-
+		L.forceMove(src)
+		L.disabilities |= MUTE
+		L.faction += "mimic" //Stops mimics from instaqdeling people in statues
+		L.status_flags |= GODMODE
 		obj_integrity = L.health + 100 //stoning damaged mobs will result in easier to shatter statues
 		max_integrity = obj_integrity
-		intialTox = L.getToxLoss()
-		intialFire = L.getFireLoss()
-		intialBrute = L.getBruteLoss()
-		intialOxy = L.getOxyLoss()
-		if(ishuman(L))
-			var/mob/living/carbon/human/H = L
-			name = "statue of [H.name]"
-			H.bleedsuppress = 1
-			if(H.gender == "female")
-				icon_state = "human_female"
-		else if(ismonkey(L))
-			name = "statue of a monkey"
-			icon_state = "monkey"
-		else if(iscorgi(L))
-			name = "statue of a corgi"
-			icon_state = "corgi"
-			desc = "If it takes forever, I will wait for you..."
-
-	if(obj_integrity == 0) //meaning if the statue didn't find a valid target
-		qdel(src)
-		return
-
-	START_PROCESSING(SSobj, src)
+		START_PROCESSING(SSobj, src)
 	..()
-	icon = L.icon
-	icon_state = L.icon_state
-	overlays = L.overlays
-	color = list(rgb(77,77,77), rgb(150,150,150), rgb(28,28,28), rgb(0,0,0))
```

Comment:
Apparently the color change don't seem to work for any mob but humans, no idea why, so I moved the code to human/petrify(). If someone knows how to make the color thing work to make monkeys all white let me know.